pkgs/development/libraries/png++/default.nix

```diff
@@ -21,21 +21,25 @@ stdenv.mkDerivation rec {
 
   postCheck = "cat test/test.log";
 
-  buildInputs = [ ]
-    ++ stdenv.lib.optional docSupport doxygen;
+  buildInputs = stdenv.lib.optional docSupport doxygen;
 
   propagatedBuildInputs = [ libpng ];
 
-  makeFlags = [ "PREFIX=\${out}" ]
-    ++ stdenv.lib.optional docSupport "docs";
+  preConfigure = stdenv.lib.optionalString stdenv.isDarwin ''
+    substituteInPlace error.hpp --replace "#if (_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && !_GNU_SOURCE" "#if (__clang__ || _POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && !_GNU_SOURCE"
+  '' + ''
+    sed "s|\(PNGPP := .\)|PREFIX := ''${out}\n\\1|" -i Makefile
```

Comment:
makeFlags are quoted, did you try this? `makeFlags = [ "PREFIX=$(out)" ];`

Reply:
I did.

Reply:
Weird, that usually works.

Reply:
To expand, I tried everything I could think of, `$(out)` in particular because it’s given as an example in the nixpkgs manual. It failed when running `make test`. I’d be interested if anybody can figure it out.